src/main/kotlin/lotto/domain/LottoStatCalculator.kt

```diff
@@ -0,0 +1,24 @@
+package lotto.domain
+
+class LottoStatCalculator(private val winningLotto: WinningLotto) {
+
+    fun getStat(lottery: Lottery): LottoStatResult = LottoStatResult(
+        lottery.groupingByRank {
+            Rank.valueOf(
+                compareWithWinningLotto(it),
+                getIsMatchBonusNumber(it)
+            )
+        }.eachCount()
+    )
+
+    private fun compareWithWinningLotto(
+        lotto: Lotto
+    ): Int {
+        return lotto.numbers.fold(0) { prev, i ->
+            if (winningLotto.contains(i)) prev + 1
+            else prev
+        }
+    }
```

Comment:
p2; 다음과 같은 형태로 줄여볼 수 있을 거 같아요.
추가로 아래 로직을 winningLotto에게 부여한다면 이 메서드도 없어도 되겠네요. 😊

```suggestion
    private fun compareWithWinningLotto(
        lotto: Lotto,
    ): Int = lotto.numbers.count { winningLotto.contains(number = it) }
```